import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/helper/DocumentMapper.java

```diff
@@ -316,5 +321,75 @@ public void setDocumentMappingStatus(Status documentMappingStatus) {
 		// TODO Auto-generated method stub
 		
 	}
+	
+	 /**
+     * Get values used in selected projects     * 
+     * @param documentMapper
+     * @return Map where the key is the field name and the value is a set of all values that are in the selected project for the field
+     */
+    public Map<String, Set<String>> getValuesUsedInSelectedProjects() {
+        valuesInSelectedProjects = new HashMap<>();
+        List<DocumentMapping> selectedActivities = getDocumentMappings().stream().filter(m -> { return Boolean.TRUE.equals(m.getSelected());}).collect(Collectors.toList());
+        
+        for (DocumentMapping docMapping : selectedActivities) {            
+            addSelectedStringMultiFieldValues(docMapping);
+            addSelectedStringFieldValues(docMapping);
+            addSelectedOrganizationFieldValues(docMapping);           
+        }
+        
+        return valuesInSelectedProjects;
+    }
 
+    private void addSelectedStringMultiFieldValues(DocumentMapping docMapping) {
+        for (Entry<String, String[]> entry : docMapping.getSourceDocument().getStringMultiFields().entrySet()) {
+            Set<String> valuesSet = valuesInSelectedProjects.get(entry.getKey());               
+            if (valuesSet == null) {
+                valuesSet = new HashSet<>();
+                valuesInSelectedProjects.put(entry.getKey(), valuesSet);
+            }                       
+            
+            String[] stringValues = (String[]) entry.getValue();
+            if (stringValues != null ) {
+                valuesSet.addAll(Arrays.asList(entry.getValue())); 
+            }               
```

Comment:
the part of adding the value to the set is the only differences, so I think you can unify the methods and have 3 method that does what from line 351354 do